R/impact_method2.R

```diff
@@ -0,0 +1,442 @@
+##' Calcualte impact with method2 that allocate impact by fvps*impact_rate
+##'
+##' @title Impact Calculation (method 2)
+##'
+##' @param con Database connection.  You will need to be \code{readonly} user
+##' to run this function.
+##' @param touchstone_name is the touchstone relevant to which impact is calculated
+##' @param cohort_min minimal birth year of interest
+##' @param cohort_max maximal birth year year of interest
+##' @param method impact calculation method - chose from method1 and method2
+##' impact outcome can be provided as age specific if simplified=FALSE
+##' @export
+impact_calculation <- function(con, touchstone_name = "201710gavi", 
+                               cohort_min = 2000, cohort_max = 2030, 
+                               routine_tot_rate_shape = "trace_cohort", method = "method2") {
+  ## prepare metadata
+  meta <- prepare_recipe(con, recipe = "impact.csv") 
+  
+  ## impact calculation
+  meta2 <- split(meta, meta$index)
+  if(method == "method1") {
+    ## method 1 
+    impact <- lapply(meta2, function(i) 
+      make_impact_method1(con, i, cohort_min, cohort_max))
+  } else {
+    ## method 2
+    impact <- lapply(meta2, function(i) 
+      make_impact(con, i, cohort_min, cohort_max, routine_tot_rate_shape))
+  }
+  
+  ## return output
+  ## meta output: group -> recipes; scripts -> calculation scripts
+  cols_group <- c("index", "touchstone", "modelling_group", "disease", "vaccine",
+                  "impact_outcome", "activity_type", "name", "focal_ingredient",
+                  "baseline", "burden_outcome")
+  
+  cols_script <- c("index", "responsibility_set_id",
+                   "responsibility_id", "burden_estimate_set_id", #"stochastic_burden_estimate_set_id",
+                   "coeff", "burden_outcome_id")
+  groups <- unique(meta[cols_group])
+  scripts <- meta[cols_script]
+  
+  ## bind impact estimates
+  message("binding data - be patient")
+  v <- impact_output(impact, method)
+  
+  return(list(groups = groups, scripts = scripts, impact_full = v$impact_full, impact_simplified = v$impact_simplified))
```

Comment:
no need for `return` here - just use `list`

Reply:
OK. Done.